apps/system/js/advanced_telemetry.js

```diff
@@ -209,20 +231,52 @@
     this.collecting = false;
     this.metrics = null;
     this.deviceID = null;
+    // Is the user idle? Updated in handleEvent() based on an idle observer
+    this.idle = false;
+    this.merge = false;
   };
 
   AT.prototype.handleEvent = function handleEvent(e) {
     switch (e.type) {
       case HUDMETRICS:
-        if (this.collecting && navigator.onLine) {
-          this.transmit(e.detail);
+        this.handleGeckoMessage(e.detail, function() {
+        });
```

Comment:
This empty function is a bit wasteful, wouldn't it be better to just not provide a callback to `handleGeckoMessage`?